### PR TITLE
feat(randomBytes): adding support to generate different bytes via RngCore

### LINKED ISSUE #8805: feat(`cheatcodes`): add `vm.random{Bytes, Bytes4, Bytes8}()`
### Component

Forge

### Describe the feature you would like

I'm trying to rewrite the FROST cryptographic protocol from Rust to Solidity, and `vm.randomBytes` would be very useful for that. I could set `fuzz.seed = '0xdeadbeef'` in Solidity, and use `StdRng::from_seed(...)` in Rust. This way I can get the same random number generator, which is very convenient for debugging and copying the implementation from Rust code.

I think it makes sense to add [`RngCore`](https://docs.rs/rand/latest/rand/trait.RngCore.html) methods as new cheat codes:
- `vm.randomBytes4` (`next_u32`)
- `vm.randomBytes8` (`next_u64`)
- `vm.randomBytes` (`fill_bytes`)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Abhishekkochar**
fixes #8805

### CODE REVIEW

File: crates/cheatcodes/src/utils.rs [hunk 1]
```diff
@@ -133,6 +133,22 @@ impl Cheatcode for randomBytesCall {
...
 
+// Random 4 bytes
+impl Cheatcode for randomBytes4Call {
+    fn apply(&self, state: &mut Cheatcodes) -> Result {
+        let rand_u32 = state.rng().next_u32();
+        Ok(B32::from(rand_u32).abi_encode())
+    }
+}
+
+// Random 8 bytes
```
Comment (DaniPopes) [inline crates/cheatcodes/src/utils.rs:144]: There is no need for these comments.

File: testdata/cheats/Vm.sol
Comment (DaniPopes) [inline testdata/cheats/Vm.sol:1]: Please revert changes to this file that are not automatically generated.

**Comment by DaniPopes** [comment]
vm.randomBytes already exists

File: crates/cheatcodes/spec/src/vm.rs [hunk 2]
```diff
@@ -2402,7 +2402,14 @@ interface Vm {
...
     #[cheatcode(group = Utilities)]
     function setArbitraryStorage(address target) external;
-}
+
+    /// Return random 4bytes
+    #[cheatcode(group = Utilities)]
+    function randomBytes4() external view returns (bytes4);
+    /// Return random 8bytes
+    #[cheatcode(group = Utilities)]
+    function randomBytes8() external view returns (bytes8);
```
Comment (DaniPopes) [inline crates/cheatcodes/spec/src/vm.rs:2411]: Please move to below `randomBytes`

File: crates/cheatcodes/spec/src/vm.rs [hunk 3]
```diff
@@ -2402,7 +2402,14 @@ interface Vm {
...
+ }
```
Comment (DaniPopes) [inline crates/cheatcodes/spec/src/vm.rs:2412]:
```suggestion
}
```

**Review by DaniPopes — CHANGES_REQUESTED** [review]

**Comment by StackOverflowExcept1on** [comment]
@DaniPopes yes, but we need new `randomBytes` that works exactly like `rng.fill_bytes` and doesn't generate random `U256`

**Review by mattsse — APPROVED** [review]
these make sense as additional utils imo

**Review by DaniPopes — APPROVED** [review]

**Pull request merged**